internal/store/api_keys_access_list.go

```diff
@@ -17,64 +17,89 @@ package store
 import (
 	"context"
 	"fmt"
+	"strings"
+
+	"github.com/mongodb/mongocli/internal/convert"
 
 	"github.com/mongodb/mongocli/internal/config"
 	atlas "go.mongodb.org/atlas/mongodbatlas"
 	"go.mongodb.org/ops-manager/opsmngr"
 )
 
-//go:generate mockgen -destination=../mocks/mock_api_keys_access_list.go -package=mocks github.com/mongodb/mongocli/internal/store OrganizationAPIKeyWhitelistLister,OrganizationAPIKeyWhitelistCreator,OrganizationAPIKeyWhitelistDeleter
+//go:generate mockgen -destination=../mocks/mock_api_keys_access_list.go -package=mocks github.com/mongodb/mongocli/internal/store OrganizationAPIKeyAccessListCreator,OrganizationAPIKeyAccessListDeleter,OrganizationAPIKeyAccessListLister
 
-type OrganizationAPIKeyWhitelistLister interface {
-	OrganizationAPIKeyWhitelists(string, string, *atlas.ListOptions) (*atlas.WhitelistAPIKeys, error)
+type OrganizationAPIKeyAccessListLister interface {
+	OrganizationAPIKeyAccessLists(string, string, *atlas.ListOptions) (*atlas.AccessListAPIKeys, error)
 }
 
-type OrganizationAPIKeyWhitelistCreator interface {
-	CreateOrganizationAPIKeyWhite(string, string, []*atlas.WhitelistAPIKeysReq) (*atlas.WhitelistAPIKeys, error)
+type OrganizationAPIKeyAccessListDeleter interface {
+	DeleteOrganizationAPIKeyAccessList(string, string, string) error
 }
 
-type OrganizationAPIKeyWhitelistDeleter interface {
-	DeleteOrganizationAPIKeyWhitelist(string, string, string) error
+type OrganizationAPIKeyAccessListCreator interface {
+	CreateOrganizationAPIKeyAccessList(string, string, []*atlas.AccessListAPIKeysReq) (*atlas.AccessListAPIKeys, error)
 }
 
-// OrganizationAPIKeys encapsulates the logic to manage different cloud providers
-func (s *Store) OrganizationAPIKeyWhitelists(orgID, apiKeyID string, opts *atlas.ListOptions) (*atlas.WhitelistAPIKeys, error) {
+// CreateOrganizationAPIKeyAccessList encapsulates the logic to manage different cloud providers
+func (s *Store) CreateOrganizationAPIKeyAccessList(orgID, apiKeyID string, opts []*atlas.AccessListAPIKeysReq) (*atlas.AccessListAPIKeys, error) {
 	switch s.service {
 	case config.CloudService:
-		result, _, err := s.client.(*atlas.Client).WhitelistAPIKeys.List(context.Background(), orgID, apiKeyID, opts)
+		result, _, err := s.client.(*atlas.Client).AccessListAPIKeys.Create(context.Background(), orgID, apiKeyID, opts)
 		return result, err
 	case config.OpsManagerService, config.CloudManagerService:
-		result, _, err := s.client.(*opsmngr.Client).WhitelistAPIKeys.List(context.Background(), orgID, apiKeyID, opts)
+		result, _, err := s.client.(*opsmngr.Client).AccessListAPIKeys.Create(context.Background(), orgID, apiKeyID, opts)
+		if err != nil {
```

Comment:
This is ok but it may get tricky, lucky for you modern go has a better approach, you can cast the generic error to the atlas error and that error has a couple of fields you can use

```go
var target *atlas.ErrorResponse
if err != nil && errors.As(err, &target) {
	if target.HTTPCode == 404 {
		// handle
	}
}
`

Reply:
for reference https://blog.golang.org/go1.13-errors